tests/testthat/test-as_kable_extra.R

```diff
@@ -1,8 +1,280 @@
-test_that("as_kable_extra() standard case", {
-  expect_error(
-    trial |>
-      tbl_summary(include = age) |>
-      as_kable_extra(format = "latex"),
-    NA
+skip_if_not(is_pkg_installed(c("kableExtra", "knitr"), reference_pkg = "gtsummary"))
+
+my_tbl_summary <- trial |>
+  select(trt, age, death) |>
+  tbl_summary()
+my_tbl_regression <- lm(marker ~ age, trial) |> tbl_regression()
+
+test_that("as_kable_extra works with standard use", {
+  # include argument does not produce warnings
+  expect_silent(my_tbl_summary |> as_kable_extra(include = tibble))
+
+  # no errors replacing default kable argument value
+  expect_silent(my_tbl_summary |> as_kable_extra(col.names = NULL))
+
+  expect_silent(kbl_summary <- my_tbl_summary |> as_kable_extra())
+
+  # correct number of rows (+2 for header styling)
+  kbl <- kbl_summary[1] |> strsplit("* </?tr>\n *")
+  expect_equal(length(kbl[[1]]), 10)
+})
+
+test_that("as_kable_extra(return_calls) works as expected", {
+  # no warnings produced
+  expect_silent(kbl <- my_tbl_summary |> as_kable_extra(return_calls = TRUE))
+
+  # correct elements are returned
+  expect_equal(
+    names(kbl),
+    c("tibble", "fmt", "cols_merge", "fmt_missing", "cols_hide", "remove_line_breaks",
+      "escape_table_body", "bold_italic", "kable", "add_indent", "footnote")
+  )
+})
+
+test_that("as_kable_extra(format) works as expected", {
+  # html (default)
+  expect_silent(kbl_html <- my_tbl_summary |> as_kable_extra())
+  kbl_html <- kbl_html[1] |> strsplit("* </?tr>\n *")
+  expect_snapshot(kbl_html)
+
+  # latex
+  expect_silent(kbl_latex <- my_tbl_summary |> as_kable_extra(format = "latex"))
+  kbl_latex <- kbl_latex[1] |> strsplit("\n")
+  expect_snapshot(kbl_latex)
+
+  # simple (warnings generated since formatting is removed)
+  expect_warning(expect_warning(
```

Comment:
I don't know why we need two, but it works 🤷🏼

Reply:
Each `expect_warning` call only catches one warning now, so 2 are needed for the 2 warnings here :)